### PR TITLE
feat: Add Azure Blob configuration to the object storage setting

### PR DESCRIPTION
**Pull request by daviderli614**
Released issue: https://github.com/GreptimeTeam/greptimedb-operator/issues/215

## Summary by CodeRabbit

- **New Features**
	- Added support for Azure Blob storage, allowing users to configure blob storage options.
	- Introduced new fields and methods for managing blob storage credentials and settings, including `AccountName` and `AccountKey`.

- **Bug Fixes**
	- Enhanced validation checks for the new Azure Blob storage provider to ensure proper configuration.

- **Documentation**
	- Updated documentation to reflect the new Azure Blob storage capabilities and configuration options.

### CODE REVIEW

File: examples/standalone/blob/standalone.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: greptime.io/v1alpha1
+kind: GreptimeDBStandalone
+metadata:
+  name: standalone-with-blob
```
Comment (zyy17) [inline examples/standalone/blob/standalone.yaml:4]: `blob` -> `azblob` will be better.

**Comment by zyy17** [comment]
@daviderli614 It's mostly LGTM. I think we should use' blob` and `AZBlob` as part of the naming instead of using `blob` directly.

**Comment by daviderli614** [comment]
> @daviderli614 It's mostly LGTM. I think we should use `azblob` and `AZBlob` as part of the naming instead of using `blob` directly.

I changed it to use azblob to define this structure:
```
  objectStorage:
    azblob:
      container: "greptimedb"
      secretName: "azblob-credentials"
      endpoint: "https://<storage-account>.blob.core.windows.net"
      root: "cluster-with-blob-data"
```

**Review by zyy17 — APPROVED** [review]
lgtm

**Pull request merged**